### ISSUE TITLE
Mysterious error due to num_workers: 1

### ISSUE THREAD

**Issue by sytelus** [pytorch/pytorch#11786]
In one of my projects I started getting this error randomly:

```
Exception ignored in: <bound method StorageRef.__del__ of <torch.multiprocessing.reductions.StorageRef object at 0x00000258213645C0>>
Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\site-packages\torch\multiprocessing\reductions.py", line 26, in __del__
AttributeError: 'NoneType' object has no attribute 'Storage'
```

One of the things I'm doing in my code is to train the model on CPU and test on GPU, so I basically do `model.to(...)` to move the model back and forth. 

**Possible Cause and Solution**

I haven't yet figured out why exactly this is happening but it turns out that test DataLoader (which I want to use on GPU), had following parameters being passed:

`'num_workers': 1, 'pin_memory': True`

After I removed `'num_workers': 1`, the error disappeared.

**Repro Code**

I haven't done the minimal version of code that reproduces this error but you can get this code from [this repo and commit](https://github.com/sytelus/NNExp/tree/f5d8cdfcde39d867cc090546a94051426d96acd8). If you run [partial_data.py](https://github.com/sytelus/NNExp/blob/f5d8cdfcde39d867cc090546a94051426d96acd8/NNExp/pytorch/mnist/partial_data.py), you will see above error shown in the console.

**Other Info**

**Comment by stephane-archer** [comment]
same problem for me when i set num_workers to something equal of bigger than 1

**Comment by peterjc123** [comment]
@sytelus Does this [post](https://github.com/pytorch/pytorch/issues/2757#issuecomment-341752166) helps?

**Comment by sytelus** [comment]
Yes, I'd looked at https://github.com/pytorch/pytorch/issues/2757#issuecomment-341752166 however that one concerns to exception occuring in `DataLoaderIter.__del__` while this one if for  `StorageRef.__del__`. Also I think former happens after program ends and GC tries to do cleanup while this one occurs while program is still running and probably more serious.

**Comment by peterjc123** [comment]
cc: @SsnL

**Comment by soumith** [comment]
cc: @ezyang didn't you fix this on master recently?
